lib/rmt/mirror/file_reference.rb

```diff
@@ -22,6 +22,11 @@ def build_from_metadata(metadata, base_dir:, base_url:, cache_dir: nil)
   def initialize(relative_path:, base_dir:, base_url:, cache_dir: nil)
     @cache_path = (cache_dir ? File.join(cache_dir, relative_path) : nil)
     @local_path = File.join(base_dir, relative_path.gsub(/\.\./, '__'))
+
+    # INFO: Encode the filename of the URI to make RMT handle not URI RFC3986 complaint filenames. (e.g. containing special characters)
```

Comment:
Not a full review, just a typo fix ;)

```suggestion
    # INFO: Encode the filename of the URI to make RMT handle RFC3986-incompliant filenames (e.g. containing special characters)
```